Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc

```diff
@@ -0,0 +1,308 @@
+  tree->Branch("t_pt", &t_pt, "t_pt/D");
+  tree->Branch("t_phi", &t_phi, "t_phi/D");
+  tree->Branch("t_mindR1", &t_mindR1, "t_mindR1/D");
+  tree->Branch("t_mindR2", &t_mindR2, "t_mindR2/D");
+  tree->Branch("t_eMipDR", &t_eMipDR, "t_eMipDR/D");
+  tree->Branch("t_eMipDR2", &t_eMipDR2, "t_eMipDR2/D");
+  tree->Branch("t_eMipDR3", &t_eMipDR3, "t_eMipDR3/D");
+  tree->Branch("t_eMipDR4", &t_eMipDR4, "t_eMipDR4/D");
+  tree->Branch("t_eMipDR5", &t_eMipDR5, "t_eMipDR5/D");
+  tree->Branch("t_eHcal", &t_eHcal, "t_eHcal/D");
+  tree->Branch("t_eHcal10", &t_eHcal10, "t_eHcal10/D");
+  tree->Branch("t_eHcal30", &t_eHcal30, "t_eHcal30/D");
+  tree->Branch("t_hmaxNearP", &t_hmaxNearP, "t_hmaxNearP/D");
+  tree->Branch("t_emaxNearP", &t_emaxNearP, "t_emaxNearP/D");
+  tree->Branch("t_eAnnular", &t_eAnnular, "t_eAnnular/D");
+  tree->Branch("t_hAnnular", &t_hAnnular, "t_hAnnular/D");
+  tree->Branch("t_rhoh", &t_rhoh, "t_rhoh/D");
+  tree->Branch("t_selectTk", &t_selectTk, "t_selectTk/O");
+  tree->Branch("t_qltyFlag", &t_qltyFlag, "t_qltyFlag/O");
+  tree->Branch("t_qltyMissFlag", &t_qltyMissFlag, "t_qltyMissFlag/O");
+  tree->Branch("t_qltyPVFlag", &t_qltyPVFlag, "t_qltyPVFlag/O");
+  tree->Branch("t_gentrackP", &t_gentrackP, "t_gentrackP/D");
+
+  t_DetIds = new std::vector<unsigned int>();
+  t_DetIds1 = new std::vector<unsigned int>();
+  t_DetIds3 = new std::vector<unsigned int>();
+  t_HitEnergies = new std::vector<double>();
+  t_HitEnergies1 = new std::vector<double>();
+  t_HitEnergies3 = new std::vector<double>();
+  t_trgbits = new std::vector<bool>();
+  tree->Branch("t_DetIds", "std::vector<unsigned int>", &t_DetIds);
+  tree->Branch("t_HitEnergies", "std::vector<double>", &t_HitEnergies);
+  tree->Branch("t_trgbits", "std::vector<bool>", &t_trgbits);
+  tree->Branch("t_DetIds1", "std::vector<unsigned int>", &t_DetIds1);
+  tree->Branch("t_DetIds3", "std::vector<unsigned int>", &t_DetIds3);
+  tree->Branch("t_HitEnergies1", "std::vector<double>", &t_HitEnergies1);
+  tree->Branch("t_HitEnergies3", "std::vector<double>", &t_HitEnergies3);
+
+  tree2 = fs->make<TTree>("EventInfo", "Event Information");
+
+  tree2->Branch("t_RunNo", &t_RunNo, "t_RunNo/i");
+  tree2->Branch("t_EventNo", &t_EventNo, "t_EventNo/i");
+  tree2->Branch("t_Tracks", &t_Tracks, "t_Tracks/I");
+  tree2->Branch("t_TracksProp", &t_TracksProp, "t_TracksProp/I");
+  tree2->Branch("t_TracksSaved", &t_TracksSaved, "t_TracksSaved/I");
+  tree2->Branch("t_TracksLoose", &t_TracksLoose, "t_TracksLoose/I");
+  tree2->Branch("t_TracksTight", &t_TracksTight, "t_TracksTight/I");
+  tree2->Branch("t_TrigPass", &t_TrigPass, "t_TrigPass/O");
+  tree2->Branch("t_TrigPassSel", &t_TrigPassSel, "t_TrigPassSel/O");
+  tree2->Branch("t_L1Bit", &t_L1Bit, "t_L1Bit/O");
+  tree2->Branch("t_allvertex", &t_allvertex, "t_allvertex/I");
+  t_hltbits = new std::vector<bool>();
+  t_ietaAll = new std::vector<int>();
+  t_ietaGood = new std::vector<int>();
+  t_trackType = new std::vector<int>();
```

Comment:
Same thing about the `new` as above, see 7.7 of https://cms-sw.github.io/cms_coding_rules.html